ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadWork.java

```diff
@@ -47,23 +51,32 @@
   final LineageState sessionStateLineageState;
 
   public ReplLoadWork(HiveConf hiveConf, String dumpDirectory, String dbNameToLoadIn,
-      String tableNameToLoadIn, LineageState lineageState)
-      throws IOException {
+      String tableNameToLoadIn, LineageState lineageState, boolean isIncrementalDump) throws IOException {
     this.tableNameToLoadIn = tableNameToLoadIn;
     sessionStateLineageState = lineageState;
     this.dumpDirectory = dumpDirectory;
-    this.iterator = new BootstrapEventsIterator(dumpDirectory, dbNameToLoadIn, hiveConf);
-    this.constraintsIterator = new ConstraintEventsIterator(dumpDirectory, hiveConf);
     this.dbNameToLoadIn = dbNameToLoadIn;
+    if (isIncrementalDump) {
+      incrementalIterator = new IncrementalLoadEventsIterator(dumpDirectory, hiveConf);
```

Comment:
Can't we create incrementalIterator inside the constructor of IncrementalLoadTasksBuilder?

Reply:
i think we should not link iterator to builder as we should be able to give different iterator to builder

Reply:
I was hoping dumpDirectory is the input which controls the different input instead of iterator as we don't have any use case for iterator outside IncrementalLoadTasksBuilder. Anyways, I'm ok with both approach.